src/tools/spake2p/Cmd_GenVerifier.cpp

```diff
@@ -143,6 +158,27 @@ uint8_t gSalt[BASE64_MAX_DECODED_LEN(BASE64_ENCODED_LEN(chip::kSpake2p_Max_PBKDF
 uint8_t gSaltDecodedLen   = 0;
 uint8_t gSaltLen          = 0;
 const char * gOutFileName = nullptr;
+FILE * gPinCodeFile       = nullptr;
+
+static uint32_t GetNextPinCode()
+{
+    if (!gPinCodeFile)
+    {
+        return chip::kSetupPINCodeUndefinedValue;
+    }
+    char pinCodeStr[9] = { 0 };
+    if (fgets(pinCodeStr, 8, gPinCodeFile) != nullptr)
```

Comment:
This does not look quite right:

1. If a line has an 8-digit number, this will only read the first 7 digits.
2. If a line has an 7-digit number, the newline will not be read, afaict, and the next read will end up with an empty string.

Reply:
Changed it to `getline`.